vscode-lean4/src/taskgutter.ts

```diff
@@ -51,30 +66,57 @@ class LeanFileTaskGutter {
 }
 
 export class LeanTaskGutter implements Disposable {
-    private decoration: TextEditorDecorationType;
-    private status: { [uri: string]: number | undefined } = {};
+    private decorations: Map<LeanFileProgressKind, [TextEditorDecorationType, string]> = new Map<LeanFileProgressKind, [TextEditorDecorationType, string]>();
+    private status: { [uri: string]: Map<LeanFileProgressKind, number> | undefined } = {};
     private gutters: { [uri: string]: LeanFileTaskGutter | undefined } = {};
     private subscriptions: Disposable[] = [];
 
     constructor(client: LeanClient, context: ExtensionContext) {
-        this.decoration = window.createTextEditorDecorationType({
-            overviewRulerLane: OverviewRulerLane.Left,
-            overviewRulerColor: 'rgba(255, 165, 0, 0.5)',
-            dark: {
-                gutterIconPath: context.asAbsolutePath('media/progress-dark.svg'),
-            },
-            light: {
-                gutterIconPath: context.asAbsolutePath('media/progress-light.svg'),
-            },
-            gutterIconSize: 'contain',
-        });
+        this.decorations.set(LeanFileProgressKind.Processing, [
+            window.createTextEditorDecorationType({
+                overviewRulerLane: OverviewRulerLane.Left,
+                overviewRulerColor: 'rgba(255, 165, 0, 0.5)',
+                dark: {
+                    gutterIconPath: context.asAbsolutePath('media/progress-dark.svg'),
+                },
+                light: {
+                    gutterIconPath: context.asAbsolutePath('media/progress-light.svg'),
+                },
+                gutterIconSize: 'contain',
+            }),
+            'busily processing...'
+        ])
+        this.decorations.set(LeanFileProgressKind.FatalError, [
+            window.createTextEditorDecorationType({
+                overviewRulerLane: OverviewRulerLane.Left,
+                overviewRulerColor: 'rgba(255, 0, 0, 0.5)',
+                dark: {
+                    gutterIconPath: context.asAbsolutePath('media/progress-error-dark.svg'),
+                },
+                light: {
+                    gutterIconPath: context.asAbsolutePath('media/progress-error-light.svg'),
+                },
+                gutterIconSize: 'contain',
+            }),
+            'processing stopped'
+        ])
 
         this.subscriptions.push(
             window.onDidChangeVisibleTextEditors(() => this.updateDecos()),
             client.progressChanged((progress) => {
                 for (const [uri, processing] of progress) {
-                    this.status[uri.toString()] = processing.length === 0 ? undefined :
-                        Math.min(...processing.map(p => p.range.start.line));
+                    if (processing.length === 0) {
+                        this.status[uri.toString()] = undefined
+                    } else {
+                        const newStatus = new Map<LeanFileProgressKind, number>()
```

Comment:
As discussed in the other issue, please just make one decoration per LeanFileProgressProcessingInfo (i.e., make LeanFileTaskGutter take a `processing: LeanFileProgressProcessingInfo[]` argument).  No need to do `Math.min`.

Reply:
I sure can do this, but this would be a greater change than just adding the red bar for fatal errors. Should this be included here or in its own PR?

Reply:
Doing it properly won't be larger than the current change.  Please include it here.

Reply:
Alright, I will do so later this week.